src/hotspot/share/opto/castnode.hpp

```diff
@@ -78,6 +78,11 @@ class CastIINode: public ConstraintCastNode {
     : ConstraintCastNode(n, t, carry_dependency), _range_check_dependency(range_check_dependency) {
     init_class_id(Class_CastII);
   }
+  CastIINode(Node* ctrl, Node* n, const Type* t, bool carry_dependency = false, bool range_check_dependency = false)
+    : ConstraintCastNode(n, t, carry_dependency), _range_check_dependency(range_check_dependency) {
+    init_class_id(Class_CastII);
+    init_req(0, ctrl);
+  }
```

Comment:
Since this node is just constraining the type range and not emitting any real hardware instruction does it really matter to tie it with an explicit control edge which will restrict GVN to share it even if Node* n is same.